eox_nelp/signals/tests/test_receivers.py

```diff
@@ -242,3 +245,177 @@ def test_alternative_mode(self, create_external_certificate_mock, generate_data_
+        """Setup common conditions for every test case"""
+        self.user, _ = User.objects.update_or_create(
+            username="Newt",
+            email="newt@example.com"
+        )
+        self.course_key = CourseKey.from_string("course-v1:test+Cx105+2022_T4")
+        profile_data = {
+            "name": "Newt Scamander"
+        }
+        setattr(self.user, "profile", set_key_values(profile_data))
+        course_enrollment_data = {
+            "user": self.user,
+            "created": timezone.now(),
+            "mode": "no-id-professional",
+            "course_id": self.course_key
+        }
+        self.course_enrollment = set_key_values(course_enrollment_data)
+
+    @override_settings(ENABLE_CERTIFICATE_PUBLISHER=False)
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_inactive_behavior(self, create_external_certificate_mock):
+        """Test that the asynchronous task wont' be called when the setting is not active.
+
+        Expected behavior:
+            - create_external_certificate is not called
+        """
+        enrollment_publisher(self.course_enrollment)
+
+        create_external_certificate_mock.delay.assert_not_called()
+
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_invalid_mode(self, create_external_certificate_mock):
+        """Test when the course enrollment has an invalid mode.
+
+        Expected behavior:
+            - create_external_certificate is not called.
+            - Invalid error was logged.
+        """
+        invalid_mode = "audit"
+        log_info = (
+            f"The {invalid_mode} enrollment associated with the user <{self.user.username}>"
+            f" and course <{self.course_key}> doesn't have a valid mode and therefore its data won't be published."
+        )
+        invalid_course_enrollment = self.course_enrollment
+
+        setattr(invalid_course_enrollment, "mode", "audit")
+
+        with self.assertLogs(receivers.__name__, level="INFO") as logs:
+            enrollment_publisher(invalid_course_enrollment)
+
+        create_external_certificate_mock.delay.assert_not_called()
+        self.assertEqual(logs.output, [
+            f"INFO:{receivers.__name__}:{log_info}"
+        ])
+
+    @patch("eox_nelp.signals.receivers.CourseGradeFactory")
+    @patch("eox_nelp.signals.receivers._generate_external_certificate_data")
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_create_call(self, create_external_certificate_mock, generate_data_mock, course_grade_factory_mock):
+        """Test when the enrollment mode is valid and the asynchronous task is called
+
+        Expected behavior:
+            - _generate_external_certificate_data is called with the right parameters.
+            - create_external_certificate is called with the _generate_external_certificate_data output.
+            - Info was logged.
+        """
+        generate_data_mock.return_value = {
```

Comment:
Also as above, this is mocked so t when this doesn't exist the process turns into an error.